WearSpeakerSample/wear/src/main/java/com/example/android/wearable/speaker/MainStateHolder.kt

```diff
@@ -16,161 +16,215 @@
 package com.example.android.wearable.speaker
 
 import android.Manifest
-import android.app.Application
+import android.app.Activity
 import android.content.pm.PackageManager
 import android.media.AudioDeviceInfo
 import android.media.AudioManager
-import androidx.constraintlayout.motion.widget.MotionLayout
+import android.media.MediaPlayer
+import androidx.annotation.RequiresPermission
+import androidx.compose.foundation.MutatorMutex
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
 import androidx.core.content.ContextCompat
 import androidx.core.content.getSystemService
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.StateFlow
-import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.suspendCancellableCoroutine
+import java.time.Duration
+import kotlin.coroutines.resume
 
 /**
- * A state holder driving the logic of [MainActivity].
+ * A state holder driving the logic of the app.
+ *
+ * This state holder is scoped to the composition (and therefore can be provided an [Activity]) to handle
+ * permission-related logic.
  */
 class MainStateHolder(
```

Comment:
This class should be called `MainState` to be consistent with Compose's naming convention